src/Web/Controllers/BookController.cs

```diff
@@ -73,4 +76,37 @@ public async Task<IActionResult> Delete([SwaggerParameter("ID of the book to del
         await this._bookService.DeleteBook(id);
         return NoContent();
     }
+
+    [HttpPost("{id}/meme")]
+    [SwaggerOperation("Creates a new book meme")]
+    [SwaggerResponse(201, "Book meme created successfully", typeof(Meme))]
+    public async Task<IActionResult> CreateBookMeme(string id)
+    {
+        var form = await Request.ReadFormAsync();
+        if (form.Files.Count == 0)
```

Comment:
You could merge this and the if statement further down I think as they both return bad request.

Reply:
I don't think we can because this is checking early if there are 0 files in the form; if i merge it with line 91 then an exception is thrown if there's no elements in the sequence